pipeline.py

```diff
@@ -72,31 +88,88 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+            article.save_as(" ".join(cleaned_tokens), kind=ArtifactType.cleaned)
+            article.save_as(" ".join(single_tagged_tokens), kind=ArtifactType.single_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        text = Mystem().analyze(raw_text)
+        tokens = []
+        for word in text:
+            if 'analysis' not in word:
+                continue
+            if not word['analysis']:
+                continue
+            morph_token = MorphologicalToken(original_word=word['text'])
+            morph_token.normalized_form = word['analysis'][0]['lex']
+            morph_token.tags_mystem = word['analysis'][0]['gr']
+
+            tokens.append(morph_token)
+
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    if not isinstance(path_to_validate, Path):
+        path_to_validate = Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    if len(list(path_to_validate.glob('*'))) == 0:
+        raise EmptyDirectoryError
+
+    counter_txt, counter_meta = 0, 0
+
+    for file in sorted(path_to_validate.glob('*'), key=lambda x: int(x.name.split('_')[0])):
```

Comment:
how do you guarantee that all files in a directory contain a number? too reckless